test/companion.jl

```diff
@@ -1,22 +1,64 @@
-n = rand(1:10)
-Z = Companion(randn(n))
+# test/companion.jl
 
-#Special properties
-@test Matrix(inv(Z)) ≈ inv(Matrix(Z))
+import LinearAlgebra: mul!
+using OffsetArrays: OffsetArray
 
-#Matvec product
-b = randn(n)
-@test Z*b ≈ Matrix(Z)*b
+v = OffsetArray(1:3, -5) # test non-1-based indexing and AbstractVector
+
+Z = @inferred Companion(v)
```

Comment:
```suggestion
Z = @inferred Companion(OffsetArrays.no_offset_view(v))
```

Reply:
Ok, this is fine with me.  I tend to agree with the decision at OffsetArrays to not implement `convert` because it is unclear what the user wants.  The `no_offset_view` is a great way to remind the code reader that we are dropping the offsets.